### ISSUE TITLE
Add tests for function matrix_dot

### ISSUE THREAD

**Issue by prithvitewatia** [prithvitewatia/matrix_h#6]

**Comment by PavlovEgor** [comment]
Hello, I noticed that the "matrix_dot" function behaves strangely even in the example from the documentation. I think the problem is in the first if (see matrix.c matrix_dot):
`if((m1.column!=1)||(m2.column!=1)||(m1.column!=m2.column)){ 
        errno=EDIMMISMATCH; 
        error(__func__); 
 }`

Perhaps the error should fall out if the number of rows is not equal to one?:
`if((m1.row!=1)||(m2.row!=1)||(m1.column!=m2.column)){
        errno=EDIMMISMATCH;
        error(__func__);
}`

**Comment by prithvitewatia** [comment]
@PavlovEgor Hi. Thanks for reporting this. I will check this and update you soon.

**Comment by prithvitewatia** [comment]
@PavlovEgor Yes, you are right. Right now, this function only supports vector multiplication. Would you like to work on this issue and make the support generic matrix dot product?
